File: frontend/src/app/models/customer.dto.ts
import { IsNotEmpty, IsPhoneNumber } from 'class-validator';

export class CustomerDto {
    id?: string;

    @IsNotEmpty()
    firstName: string = "";

    @IsNotEmpty()
    lastName!: string;

    @IsPhoneNumber()
    phoneNumber!: string;

    @IsNotEmpty()
    address!: string;

    @IsNotEmpty()
    gender!: string;

    dob?: Date;

    @IsNotEmpty()
    bankId!: string;

    openingDate?: Date;

    balance: number = 0;

    @IsNotEmpty()
    accountType!: string;

    @IsNotEmpty()
    userId!: string;
    
    constructor(){
        this.userId = ""
    }
}

File: backend/Dto/Customer.dto.cs
using System.ComponentModel.DataAnnotations;

namespace backend.Dto
{
    public class Customer
    {
        public string firstName { get; set; }
        public string lastName { get; set; }
        public int phoneNumber { get; set; }
        public string Address { get; set; }
        public string Gender { get; set; }
        public DateOnly DOB { get; set; }
        public string BankId { get; set; }
        public DateTime OpeningDate { get; set; }
        public double Balance { get; set; }
        public string AccountType { get; set; }
        public string UserId { get; set; }
    }

    public class UpdateCustomer
    {
        public string firstName { get; set; }
        public string lastName { get; set; }
        public int phoneNumber { get; set; }
        public string Address { get; set; }
        public string Gender { get; set; }
        public DateOnly DOB { get; set; }
        public string BankId { get; set; }
        public string AccountType { get; set; }
    }
}
